src/dynamic.ts

```diff
@@ -0,0 +1,75 @@
+import { EventEmitter } from 'events'
+import { FixedThreadPoolOptions, WorkerWithMessageChannel } from './fixed'
+import FixedThreadPool from './fixed'
+
+class MyEmitter extends EventEmitter {}
+
+export type DynamicThreadPoolOptions = FixedThreadPoolOptions
```

Comment:
Maybe putting type definitions is their own directory?

Reply:
I don't think that's necessary. The types fit to the classes. And just create new files just for such few lines doesn't resolve any problems IMO

Reply:
It makes the code cleaner by clearly separating types usage from types definition. And It was somehow the case before the rewrite to TS anyway. It also ease the types separation to @types package from the core features later on. Javascript only users can then have a smaller storage footprint.

Reply:
The TypeScript-specific part will end up in `d.ts` files  
If you haven't already, you should check out the branch and build it to see what happens
The `lib` folder is what actually lands in the npm package and the `src` folder is not delivered at all

So if I split the types into separate files, empty `js` files would be generated in addition to the `d.ts` files containing the types

If you really care about the bundle size, the only thing you should check out is the output of `npm publish --dry-run`